plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.ts

```diff
@@ -42,18 +45,14 @@ export class AzurePreparer implements PreparerBase {
     }
     const templateId = template.metadata.name;
 
-    const url = new URL(location); // Need to extract filepath from search parameter
     const parsedGitLocation = GitUriParser(location);
     const repositoryCheckoutUrl = parsedGitLocation.toString('https');
-
     const tempDir = await fs.promises.mkdtemp(
-      path.join(os.tmpdir(), templateId),
+      path.join(workingDirectory, templateId),
     );
 
     const templateDirectory = path.join(
-      `${path
-        .dirname(url.searchParams.get('path') || '')
-        .replace(/^\/+/g, '')}`, // Strip leading slash
+      `${path.dirname(parsedGitLocation.filepath)}`,
```

Comment:
`filepath` is available for Azure DevOps in `git-url-parse` now as well. Took the opportunity to clean that up as well.